app/Classes/Controller/MonksyncController.php

```diff
@@ -0,0 +1,576 @@
+<?php declare(strict_types = 1);
+/**
+ * @author Henrik Gebauer <henrik@mind-hochschul-netzwerk.de>, Ergänzung durch Karsten Hannig <dl1tux@t-online.de>
+ * @license https://creativecommons.org/publicdomain/zero/1.0/ CC0 1.0
+ */
+/* vim-Settings
+ * set mouse-=a
+ * set tabstop=4
+ * set shiftwidth=4
+ * set smarttab
+ */
+
+namespace App\Controller;
+
+use App\Service\Ldap;
+use Hengeb\Router\Attribute\AllowIf;
+use Hengeb\Router\Attribute\Route;
+use Symfony\Component\HttpFoundation\Response;
+
+define("MITNR", "mhnmitnr");
+
+class MonksyncController extends Controller {
+	private $tracelevel = -1;
+	private array $ldapIds =[];
+	private array $ldapUsers =[];
+	private array $monkUsers =[];
+	private array $monkLists =[];
+	private	array $stats=[];
+
+	/* 
+	 * @throws \InvalidArgumentException wrong number of arguments
+	 */ 
+	private function trace_r($tlvl,$lin, $txt,$var) {
+		if(func_num_args() !=4) {
+			throw new \InvalidArgumentException('wrong number of arguments for trace_r called in line '.$lin, __LINE__);
+			self::error(__LINE__,'wrong number of arguments for trace_r called in line '.$lin);
+		}
+		if ($this->tracelevel < $tlvl) return;
```

Comment:
Bitte return in eine eigene Zeile und in geschweiften Klammern, also so:
```
if (...) {
    return;
}
```
siehe https://www.php-fig.org/psr/psr-12/#5-control-structures